spec/outputs/graphite_spec.rb

```diff
@@ -89,122 +43,66 @@
           lines = server.pop
           expect(lines).to match(/^foo.bar.sys.data.foo 123.0 \d{10,}\n$/)
         end
-
       end
+    end
 
-      context "match all keys" do
-
-        let(:config) do <<-CONFIG
-          input {
-            generator {
-              message => "foo=123 bar=42"
-              count => 1
-              type => "generator"
-            }
-          }
-
-          filter {
-            kv { }
-          }
-
-          output {
-            graphite {
-                host => "localhost"
-                port => #{port}
-                fields_are_metrics => true
-                include_metrics => [".*"]
-                metrics_format => "foo.bar.sys.data.*"
-                debug => true
-            }
-          }
-        CONFIG
-        end
+    context "match all keys" do
 
-        let(:lines) do
-          dict = {}
-          while(!server.empty?)
-            line = server.pop
-            key  = line.split(' ')[0]
-            dict[key] = line
-          end
-          dict
-        end
+      let(:subject) { LogStash::Outputs::Graphite.new("host" => "localhost",
+                                                      "port" => port,
+                                                      "fields_are_metrics" => true,
+                                                      "include_metrics" => [".*"],
+                                                      "metrics_format" => "foo.bar.sys.data.*") }
 
-        it "match the generated foo key" do
-          expect(lines['foo.bar.sys.data.foo']).to match(/^foo.bar.sys.data.foo 123.0 \d{10,}\n$/)
-        end
+      let(:event) { LogStash::Event.new("foo" => "123", "bar" => "42") }
 
-        it "match the generated bar key" do
-          expect(lines['foo.bar.sys.data.bar']).to match(/^foo.bar.sys.data.bar 42.0 \d{10,}\n$/)
+      let(:lines) do
+        dict = {}
```

Comment:
why why not having a support method for that so the logic is more clean and packed?